include/deal.II/non_matching/quadrature_generator.h

```diff
@@ -0,0 +1,211 @@
+       * |      y         |                 (z, x)                     |
+       * |      z         |                 (x, y)                     |
+       * ---------------------------------------------------------------
+       * This is according to the convention set by the function
+       * coordinate_to_one_dim_higher.
+       */
+      template <int dim>
+      class Hypercube
+      {
+      public:
+        /**
+         * Type defining a cross section of a hypercube. The purpose of the
+         * conditional is to avoid that a 1-dimensional hypercube has a cross
+         * section. In 1D, we let the cross section be void.
+         */
+        using CrossSection =
+          typename std::conditional<(1 < dim), Hypercube<dim - 1>, void>::type;
+
+        /**
+         * Constructor, takes the center of the hypercube and its side length.
+         */
+        Hypercube(const Point<dim> &center_coordinate,
+                  const double      length_of_side);
+
+        /**
+         * Return the lower bound of the hypercube in @p direction.
+         */
+        double
+        lower_bound(const unsigned int direction) const;
+
+        /**
+         * Return the upper bound of the hypercube in @p direction.
+         */
+        double
+        upper_bound(const unsigned int direction) const;
+
+        /**
+         * Return the bounds of the hypercube in @p direction, as a one-dimensional
+         * hypercube.
+         */
+        Hypercube<1>
+        bounds(const unsigned int direction) const;
+
+        /**
+         * Returns the cross section of the hypercube orthogonal to @p direction.
+         * This is a hypercube in one dimension lower.
+         */
+        CrossSection
+        cross_section(const unsigned int direction) const;
+
+        /**
+         * Returns the point in the center of the hypercube.
+         */
+        const Point<dim> &
+        center() const;
+
+        /**
+         * Returns the length of the side of the hypercube.
+         */
+        double
+        side_length() const;
+
+        /**
+         * Returns the volume of the hypercube.
+         */
+        double
+        volume() const;
+
+        /**
+         * Returns the index:th vertex of the hypercube.
+         */
+        Point<dim>
+        vertex(const unsigned int index) const;
+
+        /**
+         * Returns the index:th child of the hypercube. Child is meant in the
+         * same way as for a cell.
+         */
+        Hypercube<dim>
+        child(const unsigned int index) const;
+
+      private:
+        /**
+         * Coordinate in the center of the hypercube.
+         */
+        const Point<dim> center_coordinate;
+
+        /**
+         * Side length of the hypercube.
+         */
+        const double length_of_side;
+      };
+
+
+      /**
+       * Returns a hypercube between 0 and 1 in all directions: [0, 1]^dim.
+       */
+      template <int dim>
+      Hypercube<dim>
+      create_unit_hypercube();
```

Comment:
It would be easier to just make these the default arguments to the constructor - what do you think?

Reply:
I made this the default of the constructor. I had to add a function to generate the default value of the `Point<dim>`. I couldn't think of another way of doing it.